src/broker_core.cpp

```diff
@@ -127,7 +128,17 @@ void broker_core::broker_init()
 	logger_->info("Initializing broker connection...");
 	workers_ = std::make_shared<worker_registry>();
 	context_ = std::make_shared<zmq::context_t>(1);
-	queue_ = std::make_shared<multi_queue_manager>();
+
+	// Yes, this may be done better, but it will do for now.
```

Comment:
If I were pedantic, I would ask you to elaborate on what's the better way. But I'm not, so keep your secrets if you want.

Reply:
You can create a static map<string, obj>, where the obj is either factory, prototype, or singleton. This map can be then filled by static constructors (so the "registration" for translation is done along with the declaration and will not be forgotten when extensions are implemented). However, for 2 or 3 manager classes, it does not worth implementing.

Reply:
heh, that sounds like so much of an overkill I'd even remove the comment

Reply:
Well, I have a personal grudge against extensive if-else branching, so I shall keep it :)